modules/dnn/src/layers/fully_connected_layer.cpp

```diff
@@ -456,6 +459,44 @@ class FullyConnectedLayerImpl CV_FINAL : public InnerProductLayer
     }
 #endif  // HAVE_INF_ENGINE
 
+
+#ifdef HAVE_INF_ENGINE
+    virtual Ptr<BackendNode> initNgraph(const std::vector<Ptr<BackendWrapper> >& inputs,
+                                        const std::vector<Ptr<BackendNode> >& nodes) CV_OVERRIDE
+    {
+        auto& ieInpNode = nodes[0].dynamicCast<InfEngineNgraphNode>()->node;
+        auto batch = ieInpNode->get_shape()[0];
+
+        std::vector<size_t> data = {(size_t)batch, (size_t)blobs[0].size[1]};
+        auto new_shape = std::make_shared<ngraph::op::Constant>(ngraph::element::i64, ngraph::Shape{2}, data.data());
+        auto inp = std::make_shared<ngraph::op::v1::Reshape>(ieInpNode, new_shape, true);
```

Comment:
Will it work with `batch size > 1`? Maybe we need reshape it to `batch x product of the rest of dims`